lib/Configuration.js

```diff
@@ -259,14 +231,45 @@ export default class Configuration {
         normalizePath(appliesFrom, this.workingDirectory)
       );
     });
-    return mergedValue(
+    return this.mergedValue(
       applyingConfigs.map((config: Object): any => config[key]),
       key,
-      this.pathToCurrentFile,
       fromFile
     );
   }
 
+  mergedValue(
+    values: Array<any>,
+    key: string,
+    fromFile: string
+  ): any {
+    let mergedResult;
+    for (let i = 0; i < values.length; i++) {
+      let value = values[i];
+      if (typeof value === 'function') {
+        value = value({
+          pathToCurrentFile: this.pathToCurrentFile,
+          pathToImportedModule: fromFile,
+          config: this,
```

Comment:
If you wanted, this function could still be a pure function with minimal fuss if you passed in `config` instead of `pathToCurrentFile`. But at this point it is starting to feel a bit potato potato.